Sources/Private/MainThread/LayerContainers/MainThreadAnimationLayer.swift

```diff
@@ -241,6 +241,17 @@ final class MainThreadAnimationLayer: CALayer, RootAnimationLayer {
     }
   }
 
+  func removeValueProvider(for keypath: AnimationKeypath) {
+    for layer in animationLayers {
+      if let foundProperties = layer.nodeProperties(for: keypath) {
```

Comment:
Like I mentioned in https://github.com/airbnb/lottie-ios/pull/2474#discussion_r1750299310, I think this behavior is different than the behavior implemented for the Core Animation rendering engine in `ValueProviderStore`.

`layer.nodeProperties(for: keypath)` returns all of the properties that match the given key path, so for example if you add a provider with `Layer.Color` and then call `removeValueProvider(for: "**.Color")`, this implementation will remove the provider for `Layer.Color` (but the implementation in the Core Animation rendering engine will not).

Reply:
I don't have a strong preference on which approach is better (removing all matches, or just removing exact matches) but I think the behavior of the Core Animation engine and Main Thread engine needs to be the same.

Reply:
I have a feeling updating the behavior of `ValueProviderStore` to check for matches (using `keypath.matches(...)` instead of `==`) would be easier than trying to change the implementation here to check for exact matches on the original input key path (which I don't think we have access to anymore).

Reply:
Thanks, @calda, for your assistance!